### DISCUSSION TITLE
Air App runs fine on Mac OS High Sierra but ask for Flash Player on Big Sur?

### DISCUSSION THREAD

**Discussion by philoupin** [airsdk/Adobe-Runtime-Support#1793] [Q&A] [answered]
Hello all,

I have a very simple Air App built via IntelliJ which contains little more than an <mx:HTML> element acting as a browser to load a URL which serves an older .swf. 
Basically this is a workaround for discontinued flash plugins not working anymore in current web browsers, so we use this Air App with an <mx:HTML> element as custom browser which only calls a single URL.

Here is the issue:
This ReplacementBrowser.app works fine when run on Mac OS High Sierra, but when run on Mac Os Big Sur (app is compiled as 64 bit app) it says: "To view this page ensure that Adobe Flash player version 10.0.0 or greater is installed."

Does anyone have an idea how to troubleshoot this situation? Obviously Flash Player is not installed anymore on the machine running OS High Sierra (or could it be?!?). I remember uninstalling Flash Player when it reached EOL.

I have the Air Runtime installed on both machines, although I would think to simply run the app (on the Big Sur machine) the Runtime wouldn't be necessary. I uninstalled the Air Runtime on the Big Sur machine and the ReplacementBrowser.app runs fine (except asking for Flash Player :) ).

I thought that an Air App doesn't need an external Flash Player to run a .swf?

Any pointers are very welcome, as I can't seem to find others reporting this problem online and I don't know where to start to track down the issue.

Many thanks!
P.

**Comment by ajwfrost** [comment, marked as answer]
Hi
It depends a little on what your HTML is doing: if this is embedding SWF content within the web page, then yes it would be loading the system-provided Flash Player binary. Back many many years ago, the Flash Player was distributed along with AIR, but - I believe because AIR then kept getting signalled as having security vulnerabilities that were purely within Flash Player - Adobe then stopped distributing the Flash Player separately within AIR, and switched those libraries to just be stub/badge libraries. The code logic is then that the AIR app looks for the Flash Player in the normal system-wide installation location, and if that isn't found then it would try loading the badge placeholder that says "Click here to install Flash Player" or similar, and of course, that's no longer possible.

One question on your Big Sur installation, is it using Apple M1 silicon?

If you're able to run your ReplacementBrowser.app on the Big Sur machine without the shared AIR runtime being installed, then it would imply that your app has got the AIR runtime already bundled within this (aka captive runtime), so if you "show contents" on the app, you can look for "Contents/Frameworks/Adobe AIR.framework" and then if you look within this and find the info.plist file, it may tell you what version of AIR was used. Deeper within this under the Resources folder there should be a "Flash Player.plugin" or similar, and if you find the actual executable binary in there, check its size. If it is small, it's the stub/badge library, if it's 10MB+ then it is actually the Flash Player and it should work (assuming it's not an M1 computer..)

To clarify the difference in AIR: it can run a SWF file that is provided at start-up, but it can't run a SWF file that's part of an HTML page. Except when using the StageWebView or HTMLLoader of course, and then you hit the limitations of the operating systems and the lack of normal support for Flash Player.

FYI we do also support the Flash Player for enterprise customers, but it's a more expensive license than AIR SDK!

thanks

**Reply by philoupin** [reply]
Thank you ver much Andrew!
I have used the latest Air distribution from Harman (33.1) to create this Air app. I am only targeting Intel, so no M1 Macs currently. I have opened the app bundle and don't seem to come across any .plist, but anyway I know what Air version I used to compile the captive runtime.
We are indeed embedding the .swf in a webpage. From what you say, the best bet then seems be to try and turn the whole .swf into an Air app, since there is no way to get the Flash player I have on High Sierra to install on Big Sur (Intel).

I have seen your offerings for support, however I am many magnitudes smaller than an "enterprise" customer, so I think this is out of my league :)

Thank you very much!
P.

**Reply by ajwfrost** [reply]
> the best bet then seems be to try and turn the whole .swf into an Air app

That would always be my preference for any of these apps that used to require the Flash Player... AIR is the way forward :-)

thanks

**Reply by philoupin** [reply]
Thanks for your advice, I will go that way then!
